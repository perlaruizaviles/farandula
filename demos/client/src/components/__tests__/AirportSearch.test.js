import React from 'react';
import {shallow} from 'enzyme';
import AirportSearch from '../Avail/AirportSearch';

describe('Rendering AirportSearch ', () => {

  const props = {
    changeSelected: () => {},
    searchChange: () => {},
    airports: [{title: 'Mexicali - MXL', description: 'General Rodolfo Se1nchez Taboada International Airport'}],
    value: "Mexicali - MXL",
    cleanField: () => {}
  };

  const tree = shallow(<AirportSearch {...props} />);

  it('Should create an snapshot for AirportSearch', () => {
    expect(tree).toMatchSnapshot();
  });

  it('Renders Search', () => {
    expect(tree.find('Search').length).toBe(1);
  });

  it('Initial Value for AirportSearch', () => {
    expect(tree.find('Search').props().value).toBe('Mexicali - MXL');
  });
});