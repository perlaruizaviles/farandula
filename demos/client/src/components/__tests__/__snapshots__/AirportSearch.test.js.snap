exports[`Rendering AirportSearch  Should create an snapshot for AirportSearch 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Search
    icon="search"
    input="text"
    minCharacters={1}
    noResultsMessage="No results found."
    onMouseDown={[Function]}
    onResultSelect={[Function]}
    onSearchChange={[Function]}
    results={
        Array [
            Object {
              "description": "General Rodolfo Se1nchez Taboada International Airport",
              "title": "Mexicali - MXL",
            },
          ]
    }
    showNoResults={true}
    style={
        Object {
            "display": "inline",
          }
    }
    value="Mexicali - MXL" />,
  "nodes": Array [
    <Search
      icon="search"
      input="text"
      minCharacters={1}
      noResultsMessage="No results found."
      onMouseDown={[Function]}
      onResultSelect={[Function]}
      onSearchChange={[Function]}
      results={
            Array [
                  Object {
                    "description": "General Rodolfo Se1nchez Taboada International Airport",
                    "title": "Mexicali - MXL",
                  },
                ]
      }
      showNoResults={true}
      style={
            Object {
                  "display": "inline",
                }
      }
      value="Mexicali - MXL" />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <AirportSearch
        airports={
                Array [
                        Object {
                          "description": "General Rodolfo Se1nchez Taboada International Airport",
                          "title": "Mexicali - MXL",
                        },
                      ]
        }
        changeSelected={[Function]}
        cleanField={[Function]}
        searchChange={[Function]}
        value="Mexicali - MXL" />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "airports": Array [
            Object {
              "description": "General Rodolfo Se1nchez Taboada International Airport",
              "title": "Mexicali - MXL",
            },
          ],
          "changeSelected": [Function],
          "cleanField": [Function],
          "searchChange": [Function],
          "value": "Mexicali - MXL",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Search
          icon="search"
          input="text"
          minCharacters={1}
          noResultsMessage="No results found."
          onMouseDown={[Function]}
          onResultSelect={[Function]}
          onSearchChange={[Function]}
          results={
                    Array [
                              Object {
                                "description": "General Rodolfo Se1nchez Taboada International Airport",
                                "title": "Mexicali - MXL",
                              },
                            ]
          }
          showNoResults={true}
          style={
                    Object {
                              "display": "inline",
                            }
          }
          value="Mexicali - MXL" />,
        "_debugID": 2,
        "_renderedOutput": <Search
          icon="search"
          input="text"
          minCharacters={1}
          noResultsMessage="No results found."
          onMouseDown={[Function]}
          onResultSelect={[Function]}
          onSearchChange={[Function]}
          results={
                    Array [
                              Object {
                                "description": "General Rodolfo Se1nchez Taboada International Airport",
                                "title": "Mexicali - MXL",
                              },
                            ]
          }
          showNoResults={true}
          style={
                    Object {
                              "display": "inline",
                            }
          }
          value="Mexicali - MXL" />,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <AirportSearch
    airports={
        Array [
            Object {
              "description": "General Rodolfo Se1nchez Taboada International Airport",
              "title": "Mexicali - MXL",
            },
          ]
    }
    changeSelected={[Function]}
    cleanField={[Function]}
    searchChange={[Function]}
    value="Mexicali - MXL" />,
}
`;
